### PR TITLE
Code cleanup for r_core_esil_step()

### PR DESCRIPTION
**Pull request by x0urc3**
**Your checklist for this pull request**
- [X] I've read the [guidelines for contributing](../DEVELOPERS.md) to this repository
- [X] I made sure to follow the project's [coding style](../DEVELOPERS.md#code-style)
- [X] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [radare2 book](../../radare2book) with the relevant information (if needed)

**Detailed description**

Observed from gdb that r_reg_set_value() for PC register is called twice during 'aes'
* remove another set PC register in initializeEsil()
* remove unnecessary call to initializeEsil() in r_core_esil_step()

**Test plan**

* Use gdb and observed that r_reg_set_value() for PC register is called once during 'aes'
* New esil test case should not fail

### CODE REVIEW

File: libr/core/cmd_anal.c [hunk 1]
```diff
@@ -4419,16 +4417,14 @@ R_API int r_core_esil_step(RCore *core, ut64 until_addr, const char *until_expr,
 	RAnalOp op = {0};
 	RAnalEsil *esil = core->anal->esil;
 	const char *name = r_reg_get_name (core->anal->reg, R_REG_NAME_PC);
+	ut64 addr;
 	bool breakoninvalid = r_config_get_i (core->config, "esil.breakoninvalid");
 	int esiltimeout = r_config_get_i (core->config, "esil.timeout");
 	ut64 startTime;
-	if (!esil) {
```
Comment (radare) [inline libr/core/cmd_anal.c]: in case esil is null, that will be crashing, if you dont run  'aeim' before 'aes' for example

Comment (x0urc3) [reply libr/core/cmd_anal.c]: There's another check down below to handle that situation
https://github.com/radareorg/radare2/blob/05ee0962800a8afa1b609e8d08ff87b3b986c357/libr/core/cmd_anal.c#L4447

Comment (x0urc3) [reply libr/core/cmd_anal.c]:
```
$> r2 /bin/ls
 -- Beer in mind.
[0x000067d0]> aes
[0x00006874]> 
```

**Comment by x0urc3** [comment]
Add test case for ```aes``` without initialization

**Pull request merged**